### PR TITLE
Selector Updating

### PR DESCRIPTION
**Pull request by sarahgos**
Have updated the Selector to work with the linked list, main menu buttons now use button nodes and games menu arrow input uses Selector class. The selection of a game still needs to be updated to run through the Selector class using Button action() (all key input should be handled by Selector class).

### CODE REVIEW

File: selector.h [hunk 1]
```diff
@@ -8,6 +8,8 @@ class Selector {
         bool first = true; 
         int up_key_state = 0;
         int down_key_state = 0;
+        int left_key_state = 0;
+        int right_key_state = 0;
```
Comment (Delcari) [inline selector.h:12]: Seems a lot of these variables registering key-states are now unused, consider cleaning up.

Comment (sarahgos) [reply selector.h]: Thanks Riley, forgot about those, have deleted now.

**Review by Delcari — APPROVED** [review]
Changes look good, happy to see the selector class handling the majority of inputs.
Well done Sarah 👏 👏

**Review by studioant — APPROVED** [review]
This is great, nice simplification of the handling. Well done Sarah!

**Review by studioant — APPROVED** [review]
Great work Sarah! The functionality is so much better now, well done.

**Pull request merged**